pkgs/pkgs-lib/formats.nix

```diff
@@ -109,18 +109,21 @@ rec {
       singleIniAtom = nullOr (oneOf [ bool int float str ]) // {
         description = "INI atom (null, bool, int, float or string)";
       };
-      iniAtom = { listsAsDuplicateKeys, listToValue }:
+      iniAtom = { listsAsDuplicateKeys, listToValue, mergeAsList }:
+        let
+          wrapper = if mergeAsList then coercedTo singleIniAtom lib.singleton else either singleIniAtom;
```

Comment:
I think renaming `wrapper` → `singleIniAtomOr` would make it a bit clearer what this wrapper function does, although it kind of obscures that it may also do coercion.